pkg/arvo/app/launch.hoon

```diff
@@ -10,77 +11,140 @@
   ==
 ::
 +$  state-zero
-  $:  tiles=(map term tile:launch)
-      tile-ordering=(list term)
-      first-time=?
+  $:  =tiles:launch
+      =tile-ordering:launch
   ==
-::
-+$  card  card:agent:gall
-++  launch-who
-  |=  =desk
-  [%pass /who %arvo %e %serve [~ /who] desk /gen/who/hoon ~]
 --
 ::
 =|  [%4 state-zero]
 =*  state  -
 %-  agent:dbug
 ^-  agent:gall
-|_  bol=bowl:gall
+|_  =bowl:gall
 +*  this  .
-    def   ~(. (default-agent this %|) bol)
+    def   ~(. (default-agent this %|) bowl)
 ::
 ++  on-init
   ^-  (quip card _this)
-  :_  this(state *[%4 state-zero])
-  [(launch-who q.byk.bol)]~
+  =/  new-state  *state-zero
+  =.  new-state
+    %_  new-state
+        tiles
+      %-  ~(gas by *tiles:launch)
+      %+  turn  `(list term)`[%chat %publish %link %dojo %weather %clock ~]
+      |=  =term
+      :-  term
+      ^-  tile:launch
+      ?+  term      [[%custom ~] %.y]
+          %chat     [[%basic 'Chat' '/~landscape/img/Chat.png' '/~chat'] %.y]
+          %links    [[%basic 'Links' '/~landscape/img/Links.png' '/~link'] %.y]
+          %dojo     [[%basic 'Dojo' '/~landscape/img/Dojo.png' '/~dojo'] %.y]
+          %publish
+        [[%basic 'Publish' '/~landscape/img/Publish.png' '/~publish'] %.y]
+      ==
+    ==
+  [~ this(state [%4 new-state])]
 ::
 ++  on-save  !>(state)
-::
 ++  on-load
   |=  old=vase
   ^-  (quip card _this)
   =/  old-state  !<(versioned-state old)
   ?:  ?=(%4 -.old-state)
     [~ this(state old-state)]
-  :_  this
-  ::%+  weld
-  ::  []~  ::  TODO: kill all subscriptions
-  :~  (launch-who q.byk.bol)
-      [%pass / %arvo %e %disconnect [~ /]]
-  ==
+  =/  new-state  *state-zero
+  =.  new-state
+    %_  new-state
+        tiles
+      %-  ~(gas by *tiles:launch)
+      %+  turn  `(list term)`[%chat %publish %link %dojo %weather %clock ~]
+      |=  =term
+      :-  term
+      ^-  tile:launch
+      ?+  term      [[%custom ~] %.y]
+          %chat     [[%basic 'Chat' '/~landscape/img/Chat.png' '/~chat'] %.y]
+          %links    [[%basic 'Links' '/~landscape/img/Links.png' '/~link'] %.y]
+          %dojo     [[%basic 'Dojo' '/~landscape/img/Dojo.png' '/~dojo'] %.y]
+          %publish
+        [[%basic 'Publish' '/~landscape/img/Publish.png' '/~publish'] %.y]
+      ==
+    ==
+  :_  this(state [%4 new-state])
+  :-  [%pass / %arvo %e %disconnect [~ /]]
+  %+  turn  ~(tap by wex.bowl)
+  |=  [[=wire =ship =term] *]
+  ^-  card
+  [%pass wire %agent [ship term] %leave ~]
 ::
 ++  on-poke
-  |=  [mar=mark vas=vase]
+  |=  [=mark =vase]
   ^-  (quip card _this)
-  ?+    mar  (on-poke:def mar vas)
-      %json
-    ?>  (team:title our.bol src.bol)
-    =/  jon  !<(json vas)
-    :-  ~
-    ?.  =(jon [%s 'disable welcome message'])
-      this
-    this(first-time %.n)
-  ==
```

Comment:
I'm not seeing this replicated in the rewrite — I assume the `first-time` stuff is going to be put in later?